### PR TITLE
OPSEXP-2300: align alfresco-sync-service charts to latest conventions

### PR DESCRIPTION
**Pull request by alxgomz**
Ref: OPSEXP-2300

* adds support for message broker existing configmap
* adds support for message broker existing secret
* adds support for db existing configmap
* adds support for db existing secret
* adds support for repo existing configmap
* fix named templates
* remove recursive usage of  `nameOverride`
* add service account
* fix resource naming

### CODE REVIEW

File: charts/alfresco-sync-service/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        {{- end }}
+      annotations:
+        {{- with .Values.podAnnotations }}
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- include "alfresco-common.component-pod-security-context" .Values | indent 4 }}
+      serviceAccountName: {{ include "alfresco-sync-service.serviceAccountName" . }}
+      {{- include "alfresco-common.imagePullSecrets" . | indent 6 }}
+      terminationGracePeriodSeconds: 60
```
Comment (gionn) [inline charts/alfresco-sync-service/templates/deployment.yaml:32]: docker image is not handling correctly shutdown requests?

worth having a value to configure it imho

File: charts/alfresco-sync-service/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      - name: syncservice
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- include "alfresco-common.component-security-context" .Values | indent 6 }}
+        envFrom:
+        - configMapRef:
+            name: {{ template "alfresco-sync-service.fullname" . }}
+        env:
+          {{- with .Values.database }}
+          {{- $dbCtx := dict "Values" (dict "nameOverride" "alfresco-sync-database") "Chart" $.Chart "Release" $.Release }}
```
Comment (gionn) [inline charts/alfresco-sync-service/templates/deployment.yaml:43]: I don't like having the duplication of the nameOverride value, better have a dedicated named template (e.g. `.name` in _helpers-name.tpl)

File: charts/alfresco-sync-service/templates/ingress.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 apiVersion: {{ template "common.capabilities.ingress.apiVersion" . }}
 kind: Ingress
 metadata:
-  name: {{ template "alfresco-sync-service.fullname" . }}-ingress
+  name: {{ template "alfresco-sync-service.fullname" . }}
   annotations:
     kubernetes.io/ingress.class: "nginx"
```
Comment (gionn) [inline charts/alfresco-sync-service/templates/ingress.yaml:8]: check the helm install logs, I am switching to `ingressClassName: nginx` in search service PR

File: charts/alfresco-sync-service/templates/ingress.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@
 apiVersion: {{ template "common.capabilities.ingress.apiVersion" . }}
```
Comment (gionn) [inline charts/alfresco-sync-service/templates/ingress.yaml:3]: I guess we can get rid of this

File: charts/alfresco-sync-service/templates/deployment.yaml (hunk 1 above)
Comment (alxgomz) [reply charts/alfresco-sync-service/templates/deployment.yaml:32]: I'm not sure tbh. I've just kept what was there before. But after taking a quick look it seems the docker image is using a wrapper script that does some substitutions before spawning the server.... So I can only imagine that this is indeed not handled properly by docker

File: charts/alfresco-sync-service/values.yaml [hunk 5]
```diff
@@ -54,83 +54,79 @@ readinessProbe:
...
+  password: null
+  existingConfigMap:
+    name: null
+    keys:
+      # -- configmap key where to find the URL of the database
+      url: DATABASE_URL
+      # -- configmap key where to find the JDBC driver class to use.
+      # The configmap may leverage the alfresco-repository.db.cm named
+      # template to auto-generate it from the sole url parameter.
+      driver: DATABASE_DRIVER
```
Comment (gionn) [inline charts/alfresco-sync-service/values.yaml:112]: not sure it make sense to have driver here but I understand is correlated to url so if it change then you may have to provide a different driver

**Review by gionn — APPROVED** [review]

**Pull request merged**